src/Components/Dropdown.tsx

```diff
@@ -0,0 +1,112 @@
+import { useMemo, useState } from 'react';
+import { peopleFromServer } from '../data/people';
+import { Person } from '../types/Person';
+import debounce from 'lodash.debounce';
+import cn from 'classnames';
+
+type Props = {
+  onSelected: (person: Person | null) => void;
+  delay: number;
+};
+
+export const Dropdown: React.FC<Props> = ({ onSelected, delay }) => {
+  const [isDropdownActive, setIsDropdownActive] = useState(true);
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+
+  // const applyQuery = useCallback(debounce(setAppliedQuery, 300), []);
+  // useCallback(() => debounce(SetAppliedQuery, 300), )
```

Comment:
Please, do not leave comments next time
If you have some doubts about code or have questions it's better to ask in the chat or at least leave comment on the GitHub